### PR TITLE
Image compression for show and user pics

### PR DESCRIPTION
**Pull request by binerys**
Use's [Jimp](https://github.com/oliver-moran/jimp) to resize uploaded images to a width of 512 and reduce image quality by 20%. Fixes issue #54 
For example, the following image: 
image

is reduced to:
image

Since image compression takes a little longer, image uploading is not as fast as it used to be. This might cause some users to think their image upload failed. Might be a good idea to include some type of spinner to indicate the image is being uploaded in the future.

### CODE REVIEW

**Comment by binerys** [comment]
Added a message regarding recommended image width in panel:

image
image

File: routes/panel.js [hunk 1]
```diff
@@ -9,189 +9,194 @@ var shows = require('../database/shows');
 var faqs = require('../database/faqs');
 var messages = require('../database/messages');
 
+// Image compression module
+var Jimp = require('jimp');
```
Comment (nathunsmitty) [inline routes/panel.js]: It'd be nice if this was also `const`. Actually, const everywhere for variables that aren't modified are 👍

File: routes/panel.js [hunk 2]
```diff
@@ -209,361 +214,389 @@ var freeChat = function(req, res) {
...
+
+        // Perform image compression on JPG/PNG only
+        if (mimeType == "image/jpeg" || mimeType == "image/png") {
+          Jimp.read(imgPath, function(err, img) {
+            if (err) { errorCallback(err); }
+            img.resize(RESIZE_WIDTH, Jimp.AUTO).quality(IMAGE_QUALITY).write(imgPath);
+          });
+        }
+        else {
+          console.log("Non-png/jpg file type");
```
Comment (nathunsmitty) [inline routes/panel.js]: Do we maybe want to do something here? I know other file formats have caused us trouble in the past, perhaps we should only accept jpegs and pngs?

**Comment by nathunsmitty** [comment]
Also nice job converting all whitespace to spaces lol

File: routes/panel.js (hunk 2 above)
Comment (binerys) [inline routes/panel.js]: Currently the `InputFileUpload` component has a property that lets you specify which file types will be accepted. We currently accept 'png/jpg/gif' which is why I added that check. However, the accepted file property can still be circumvented so I think in the future would should cause our API to return a 403 Forbidden.

**Pull request merged**